src/ert/shared/hook_implementations/forward_model_steps.py

```diff
@@ -0,0 +1,574 @@
+""",
+        )
+
+
+class DeleteDirectory(ForwardModelStepPlugin):
+    def __init__(self) -> None:
+        super().__init__(
+            name="DELETE_DIRECTORY",
+            command=[
+                os.path.abspath("../share/ert/shell_scripts/delete_directory.py"),
+                "<DIRECTORY>",
+            ],
+        )
+
+    @staticmethod
+    def documentation() -> Optional[ForwardModelStepDocumentation]:
+        return ForwardModelStepDocumentation(
+            category="utility.file_system",
+            description="""
+The :code:`DELETE_DIRECTORY` job will recursively remove a directory
+and all the files in the directory. Like the :code:`DELETE_FILE` job
+it will *only* delete files and directories which are owned by the
+current user. If one delete operation fails the job will continue, but
+unless all delete calls succeed (parts of) the directory structure
+will remain.
+
+.. warning::
+  If the directory to delete is a symlink to a directory, it will only delete
+  the link and not the directory. However, if you add a trailing slash to the
+  directory name (the symlink), then the link itself is kept, but the directory
+  it links to will be removed.
+
+""",
+            examples="""
+.. code-block:: bash
+
+    FORWARD_MODEL DELETE_DIRECTORY(<DIRECTORY>=path/to/directory)
+""",
+        )
+
+
+class DeleteFile(ForwardModelStepPlugin):
+    def __init__(self) -> None:
+        super().__init__(
+            name="DELETE_FILE",
+            command=[
+                os.path.abspath("../share/ert/shell_scripts/delete_file.py"),
+                "<FILES>",
+            ],
+        )
+
+    @staticmethod
+    def documentation() -> Optional[ForwardModelStepDocumentation]:
+        return ForwardModelStepDocumentation(
+            category="utility.file_system",
+            description="""
+The :code:`DELETE_FILE` job will *only* remove files which are owned
+by the current user, *even if* file system permissions would have
+allowed the delete operation to proceed. The :code:`DELETE_FILE` will
+*not* delete a directory, and if presented with a symbolic link it
+will *only* delete the link, and not the target.
+""",
+            examples="""
+            .. code-block:: bash
+
+                FORWARD_MODEL DELETE_FILE(<FILES>=path/file path/file2 path/fileN)
+""",
+        )
+
+
+class Eclipse100(ForwardModelStepPlugin):
+    def __init__(self) -> None:
+        super().__init__(
+            name="ECLIPSE100",
+            command=[
+                os.path.abspath("../share/ert/forward-models/res/script/ecl100.py"),
+                "<ECLBASE>",
+                "-v",
+                "<VERSION>",
+                "-n",
+                "<NUM_CPU>",
+                "<OPTS>",
+            ],
+            default_mapping={"<NUM_CPU>": 1, "<OPTS>": ""},
+        )
+
+    def validate_pre_experiment(self, fm_step_json: ForwardModelStepJSON) -> None:
+        if "version" not in self.private_args:
```

Comment:
Could you add a quick unit test for this check? Otherwise, everything looks great :)

Reply:
I did add it, then I removed it because there was already one there; `test_that_eclipse100_require_version_field():`, it is now changed to expect the raised error from the plugin forward model